samples/subsys/display/lvgl/src/main.c

```diff
@@ -20,6 +21,10 @@ LOG_MODULE_REGISTER(app);
 
 static uint32_t count;
 
+#if DT_NODE_EXISTS(DT_ALIAS(pwm_lcd0))
+static const struct pwm_dt_spec pwm_lcd0 = PWM_DT_SPEC_GET(DT_ALIAS(pwm_lcd0));
```

Comment:
I'd use a chosen rather than an alias, then it goes 1:1 with the display, something like this maybe?

```
		zephyr,display = &gc9a01;
		zephyr,display-backlight-pwm = &pwm_lcd0;
```

Reply:
Well in this case everything should be converted to something like `zephyr,xxx` ? I based this proposal on leds/buttons that are simply something like `sw0 = &button0;`

By the way if we do that, then you can get a device with `DEVICE_DT_GET(DT_CHOSEN(zephyr_display_backlight_pwm))` I guess. I it really wanted to consider the backlight like a "device" ? It's not really one according to me...

Reply:
Not sure I get your point on "get a device" it's the same thing you'd just do `PWM_DT_SPEC_GET(DT_CHOSEN(zephyr_display_backlight_pwm));`, also I would use a separate define for the node so that you can use it in the various `DT_NODE_EXISTS`

```c
#define BACKLIGHT_PWM_NODE DT_CHOSEN(zephyr_display_backlight_pwm)
...
static const struct pwm_dt_spec pwm_lcd0 = PWM_DT_SPEC_GET(BACKLIGHT_PWM_NODE);
...
#if DT_NODE_EXISTS(BACKLIGHT_PWM_NODE)
...
etc
```

not sure about the alias vs chosen, they look interchangeable to me but I feel like this node should sit next to the display one. @jfischer-no any opinion? I don't feel strongly about it btw it just looks like it makes sense to me.

Reply:
In my opinion, this PR hides issues that should be fixed/discussed in a separate PR. A PR like "Add support for the foobar board" should only be about the board itself. Anything that fixes/touches samples/subsystems/drivers should be moved to a separate PR.

Reply:
Are you asking to split the sample PWM backlight out in its own PR? I don't disagree...

Reply:
I don t disagree as well. Do you want 2 parallel PR ? The name of the node is dependent for each other.
What about the first commits about the touchscreen ic id and the bug fix on the display driver ? Separated or not ?

Reply:
It will be also easier to review each topic separately for sure :-) initially my intention was just to have the board support here, but one by one I discovered issues by checking with the samples :-)

Reply:
@joelguittet Looks like your first three commits could be merged very quickly. Your board commit looks good. So yes, please split it into two or three PRs, whatever you prefer.

Reply:
Good, will do ! After the fosdem this weekend :-) so should be done next week

Reply:
FYI RC1 is going to be cut next Monday I think so it's either now or after the release (Feb 25th).